crates/context_aware_config/src/api/type_templates/types.rs

```diff
@@ -20,3 +22,33 @@ pub struct QueryFilters {
     pub count: Option<i64>,
     pub page: Option<i64>,
 }
+
+#[derive(Debug, Deserialize, Serialize, AsRef, Deref, DerefMut, Into, Clone)]
```

Comment:
These derive macro should be used as per needed and we should not have all the macros for multiple types , as this will increase compilation time , also adding or removing wouldn't be straight forward in that case